File: lab5(polymorphism)/py/main.py
from foo import *
n = 3
m =4
arr = create_arr(n, m)
out_vectors(arr ,n, m)
sum_of_par , par = sum_of_Paralel(arr, n)
sum_of_per  , per = sum_of_Perpen(arr ,n , m)
print(f"sum of parallel vectors to 0-vec R^2: {sum_of_par}")
print(f"sum of parallel vectors to 0-vec R^3: {sum_of_per}")
out_to_zero(par ,"All vectors that paral. to 0-vec :")
out_to_zero(per , "All vectors that perpen. to 0-vec :")

File: lab5(polymorphism)/cpp/main.cpp
#include <iostream>
#include "foo.h"
int main () {
    vector<TVector *> arr;
    vector<TVector *> paral;
    vector<TVector *> perpen;
    create_arr(arr , 3, 4);
    out_All(arr ,3 ,4);
    cout << "sum of parallel vectors to 0-vec :"<<sum_of_Paralel(arr, 3 , paral)<<endl;
    cout << "sum of perpendicular vectors to 0-vec:"<<sum_of_Perpen(arr ,3,4 ,perpen)<<endl;
    out_to_zero(paral ,"All vectors that paral. to 0-vec :");
    out_to_zero(perpen , "All vectors that perpen. to 0-vec :");
}